janus_core/calculations/eos.py

```diff
@@ -0,0 +1,206 @@
+"""Equation of State."""
+
+from logging import Logger
+from typing import Any, Optional
+
+from ase import Atoms
+from ase.eos import EquationOfState
+from ase.units import kJ
+from numpy import float64, linspace
+from numpy.typing import NDArray
+
+from janus_core.calculations.geom_opt import optimize
+from janus_core.helpers.janus_types import EoSNames, EoSResults, PathLike
+from janus_core.helpers.log import config_logger
+from janus_core.helpers.utils import none_to_dict
+
+
+def _calc_volumes_energies(
+    struct: Atoms,
+    min_lattice: float = 0.95,
+    max_lattice: float = 1.05,
+    n_lattice: int = 7,
+    minimize_all: bool = False,
+    minimize_kwargs: Optional[dict[str, Any]] = None,
+    logger: Optional[Logger] = None,
+) -> tuple[NDArray[float64], list[float], list[float]]:
+    """
+    Calculate volumes and energies for all lattice constants.
+
+    Parameters
+    ----------
+    struct : Atoms
+        Structure.
+    min_lattice : float
+        Minimum lattice constant scale factor. Default is 0.95.
+    max_lattice : float
+        Maximum lattice constant scale factor. Default is 1.05.
```

Comment:
These are strictly the `min_volume`/`max_volume` given they're rescaled by `^1/3` unless I'm mistaken.